### ISSUE TITLE
Normalizing/Scaling  for PCA initialization with mexican hat function

### ISSUE THREAD

**Issue by carocamargo** [JustGlowing/minisom#118]
Hi, I'm trying to use minisom to obtain sea-level variability clusters. The use of SOM in oceanographic studies had been discussed in some articles, where they describe the best parameters, but most of them make use of the MatLab SOM toolbox. 
All papers suggest a linear initialisation (to which the PCA initialization of minisom would be equivalent), and the 'ep' neighbourhood function (which I believe is equivalent, or at least most similar to, to the mexican_hat function). 

Before performing SOM, the data requires some treatment:
1. It must be concatenated from 3D (time,lat,lon) to 2D (time, latxlon)
2.  All NaNs must be removes
3. Data must be normalized. 

When I normalize the dataset with the `sklearn.preprocessing.MinMaxScaler` function, 
![image](https://user-images.githubusercontent.com/47150926/135104293-8b1fa019-f88e-4d10-bb29-2f281f28b69e.png)

and try the PCA initialization with the gaussian function, it seems to work:
![image](https://user-images.githubusercontent.com/47150926/135102991-9e07143c-0283-4bfe-8514-8d67beb4fc34.png)

But if I try with the mexican_hat function, it doesn't work:
![image](https://user-images.githubusercontent.com/47150926/135102868-42ad8306-76ae-4701-9dcc-d5a51c17dfe1.png)

if I change to random initialization, and keep the mexican hat, then it works again:
![image](https://user-images.githubusercontent.com/47150926/135103256-29df26be-8c46-46a6-bbae-5ac71e3438d2.png)

If instead of normalizing the dataset (so that it ranges from 0 to 1), I scale it... 
![image](https://user-images.githubusercontent.com/47150926/135103824-4b52c070-b345-4c6c-917f-cebf97ae541d.png)

and then try the PCA initialization, wether with the gaussian or mexican hat, it works, but it gives very large QE.
![image](https://user-images.githubusercontent.com/47150926/135103885-729519ac-ee84-4534-a522-070bc1f730c5.png)

Can you explain to me why this is happening and what is the correct way of dealing with this?
I apologize if this is something trivial, but this is my first contact with machine learning/neural network methods. 

Thanks
[SOm-testing.pdf](https://github.com/JustGlowing/minisom/files/7244752/SOm-testing.pdf)

**Comment by JustGlowing** [comment]
hi there,

While MixMax can work in some cases, standard scaling produces better results most of the time (keep in mind that PCA assumes that the data is scaled). I'd say that you found a sensible way to scale your data.

Regarding the quantization error, it's larger with standard scaling because it allows larger values in your data.

Btw, cool charts!

**Comment by carocamargo** [comment]
Hi. 
Thanks for the quick feedback. 
Right.. so it's ok then that the QE is so large? Or does that mean that my clustering is not good? 
I've seen references where people find the parameters to have QE as low as 0.1, but mine doesn't seem to go so low...
I'm still trying to a way of defining the best sigma and learning rate...

**Comment by JustGlowing** [comment]
Regarding the quantization error (QE), read again my message above. It really depends on the data that you have and how you normalize it.

The topic of picking the best parameters has been discussed in previous issues. Please have a look.

**Comment by carocamargo** [comment]
Thanks @JustGlowing.
Do you have an example of how to plot the learning curve with batch training? 
I tried just replacing the `rand_i` for `i` in the `Basic_usage` example, but it doesn't seem to be the correct way. Thanks

**Comment by JustGlowing** [comment]
you can use the following index `sequential_index = i % numeber_of_samples-1`

**Comment by carocamargo** [comment]
> you can use the following index `sequential_index = i % numeber_of_samples-1`

Thanks, this seems to work.

I have another question... I was under the impression that the 'mexican_hat function' was somehow equivalent to the 'ep' function of the [MatLab SOM toolbox](http://www.cis.hut.fi/somtoolbox/) (which has been [suggested](https://agupubs.onlinelibrary.wiley.com/doi/10.1029/2005JC003117) to be the function that gives the least smoothing to the SOM units , but now I'm thinking they are different things. 
So, I've been trying to do some tests trying to find the ideal function to use (by changing the neighborhood function in the ['color space mapping' example](https://github.com/JustGlowing/minisom/blob/master/examples/ColorSpaceMapping.ipynb). 
Am I correct to say that the 'bubble' function is the one with the least smoothing effect? Thanks!

**Comment by JustGlowing** [comment]
Yes indeed, the bubble function is discounts at the borders.
